### ISSUE TITLE
Split-CTM returns rank-1 corners on a physical SU state (and a test documents this as expected)

### ISSUE THREAD

**Issue by yingjerkao** [tenax-lab/tenax#726]
> 🤖 **AI-generated issue** — written by Claude Code, posted by @yingjerkao.

Found while answering the open scope question in #723. Filing separately because it is the split-CTM path (#463 territory), not the fused single-site path #723 is about.

## Observation

On a **physical** simple-update state (D=2 sublattice-rotated Heisenberg, 60 SU steps), `ctm_split_tensor_converge_explicit` returns rank-1 corners:

```
split-CTM chi=4: corner ranks [1, 1, 1, 1]
split-CTM chi=8: corner ranks [1, 1, 1, 1]
```

(40 sweeps, `warmup_steps=2`; rank counted as singular values above `1e-10 · s[0]`.)

## Why that is not physics

The **same** `A`, same χ, through the 2x2 multisite recipe has an ordinary corner spectrum:

| recipe | C1 singular values at χ=8 (normalised) | rank |
|---|---|---|
| 2x2 multisite | `1, 0.128, 0.127, 0.0164, 2.08e-3, 2.02e-3` | 6 |
| split-CTM | `1, 0, 0, 0, 0, 0` | 1 |

The degenerate pairs are what you would expect from this state. A rank-1 corner is a χ_eff = 1 boundary — a mean-field environment, not the corner transfer matrix of a correlated iPEPS.

## Why this matters beyond the corner itself

`tests/test_regularized_svd.py` (in `test_split_ctm_explicit_gradient_finite`, comment at lines ~244-250) currently documents the rank-1 corner as **expected behaviour**, and shapes a gradient test around it:

```python
# NOTE: do *not* use ``sum(C1**2)`` here. The #463 double-layer
# split path renormalizes each corner per move (max_abs_normalize),
# and the boundary corner of a uniform 1-site iPEPS is genuinely
# rank-1, so the normalized C1 collapses to a constant one-hot
# matrix that is identically independent of A — its gradient is
# exactly zero. The A dependence lives in the edge tensors, so sum
# over the full environment to exercise the explicit-AD gradient.
```

"the boundary corner of a uniform 1-site iPEPS is genuinely rank-1" is the claim contradicted by the table above. If the rank-1 corner is a bug rather than physics, then the test is not merely mis-commented — it has been rewritten to route *around* the symptom, so the corner contribution to the explicit-AD gradient is currently untested on this path.

## What is not established here

I did **not** trace the split-CTM mechanism. In the fused single-site path the cause is known (#723): the Fishman cross-product `M = C1g† C4g` is indexed by the outer χ legs only, so the `chi_init=1` rank-1 corner is an exact fixed point of the corner update. The split path builds its projectors differently, so the same mechanism is *plausible but unverified* here. It could equally be a distinct bug that produces the same signature.

Worth checking whether the split path shares the "outer-χ-only cross product" structure, and if so whether the same fix (seed χ_eff = D², or retain the fused index in the cross product) applies.

## Reproducer

```python
import jax, jax.numpy as jnp
jax.config.update("jax_enable_x64", True)
from tenax.algorithms.ad_utils import ctm_split_tensor_converge_explicit as conv
from tenax.algorithms.ipeps import heisenberg_gate, ipeps, sublattice_rotate_gate
from tenax.algorithms.ipeps_config import CTMConfig, iPEPSConfig

g = sublattice_rotate_gate(heisenberg_gate())
cfg = iPEPSConfig(max_bond_dim=2, num_imaginary_steps=60, dt=0.05,
                  ctm=CTMConfig(chi=8, max_iter=20))
_, (A, B), _ = ipeps(g, None, cfg)

for chi in (4, 8):
    env = conv(A, chi=chi, num_steps=40, warmup_steps=2)
    ranks = []
    for n in ("C1", "C2", "C3", "C4"):
        s = jnp.linalg.svd(jnp.asarray(getattr(env, n).todense()), compute_uv=False)
        ranks.append(int(jnp.sum(s > 1e-10 * s[0])))
    print(f"split-CTM chi={chi}: corner ranks {ranks}")
```

Compare against `ctm_tensor_2site(A, A, chi=chi, max_iter=150, conv_tol=1e-12)` on the same `A` for the reference spectrum.

Related: #723 (same signature, fused single-site path, root cause known), #463, #666.

**Comment by yingjerkao** [comment]
> 🤖 **AI-generated comment** — written by Claude Code, posted by @yingjerkao.

Independent confirmation, measured during #921's review. Still open — this is data, not a fix.

The split `1x1` collapse is **quantitatively identical to the fused one**, on the D=2 Heisenberg SU state at `conv_tol=1e-12, max_iter=100`:

```
recipe  chi  rank   energy
   1x1    4     1   -0.649578563296
   1x1    8     1   -0.649578563296
   1x1   16     1   -0.649578563296
   1x1   32     1   -0.649578563296
   2x2    4     4   -0.658565991677
   2x2    8     8   -0.657831430578
   2x2   16    16   -0.657820225418
   2x2   32    25   -0.657821074222
```

Corner rank 1 at every χ, energy **bit-identical to 12 digits across an 8× change in χ**, and 8.2e-3 below the full-rank answer. That is the #723/#726/#747 signature exactly, on the split path, through the same legacy projector.

## Why this is being posted here

I initially scoped the split entry points *out* of #921's deprecation, on the grounds that #911's measurement was fused-path only and I had no equivalent for split. Codex review pushed back — these are exported public API reusing the same collapsing projector — so I measured instead of arguing. The numbers above are why `ctm_split_tensor` and `ctm_split_tensor_2site` now warn.

## What it does and does not settle

It settles that the split collapse is the same defect, not a separate one, and that "a test documents this as expected" is documenting a bug. It does **not** fix the projector.

Once #921's phase 1 lands, a caller hitting this gets a `DeprecationWarning` naming the migration target rather than a silent χ-independent mean-field energy — which is mitigation, not resolution. Phase 2 (rejection) is what would let this close.

**Comment by yingjerkao** [comment]
> 🤖 **AI-generated comment** — written by Claude Code, posted by @yingjerkao.

Closing as fixed. The default-path collapse is resolved: `ctm_tensor` now defaults to the 2x2 recipe (988c2a8, #723) and the single-site split-CTM path routes through 2x2 (d316ef8, #746), so a physical SU state gets full-rank corners instead of rank-1. Guarded by `test_split_ctm_corner_gradient_is_not_zero` (tests/test_regularized_svd.py:278), which asserts a zero corner gradient on the 1x1 recipe and a nonzero one (~3.26) on 2x2.

Residual by design: the now-deprecated `recipe="1x1"` still collapses to rank-1 corners and emits a DeprecationWarning (42aef1f, #911/#921). That path is tracked for removal by #911, not by this issue.
